calculator_test.go

```diff
@@ -1,24 +1,173 @@
-package calculator_test
+package calculator
 
 import (
-	"calculator"
+	"math"
 	"testing"
 )
 
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	var testCases = []struct {
+		a    float64
+		b    []float64
+		want float64
+		name string
+	}{
+		{20, []float64{1}, 21, "Adding two positive numbers yields a positive number"},
+		{231, []float64{56}, 287, "Adding three positive numbers yields a positive number"},
+		{79, []float64{123191023123}, 123191023202, "Adding one large and one small number"},
+		{0, []float64{0}, 0, "Adding two zeros yields zero"},
+		{1.2, []float64{333}, 334.2, "Adding a number with a fraction and a number without one yields a number with fraction"},
+		{-73, []float64{99}, 26, "Adding a negative number and a positive number"},
+		{-1, []float64{-1}, -2, "Adding two negative numbers yields a negative number"},
+		{-2222, []float64{0}, -2222, "Adding a number to zero yields that number"},
+		{0.0000005133, []float64{0.00000312, 0.000664, 0.0000532, 0.00001}, 0.0007308333, "Adding five very small numbers"},
+	}
+	for _, c := range testCases {
+		got := Add(c.a, c.b...)
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.name, c.want, got)
+		}
 	}
 }
 
-// func TestSubtract(t *testing.T) {
-// 	t.Parallel()
-// 	var want float64 = 2
-// 	got := calculator.Subtract(4, 2)
-// 	if want != got {
-// 		t.Errorf("want %f, got %f", want, got)
-// 	}
-// }
+func TestSubtract(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		a    float64
+		b    []float64
+		want float64
+		name string
+	}{
+		{1, []float64{1}, 0, "Subtracting a positive number from itself yields zero"},
+		{-51, []float64{-51}, 0, "Subtracting a negative number from itself yields zero"},
+		{4323, []float64{12, 11, 39}, 4261, "Subtracting 3 positive number from a much larger positive number yields a positive number"},
+		{11.2, []float64{89.8}, -78.6, "Subtracting a positive number from a smaller positive number yields a negative number"},
+		{-100, []float64{-111}, 11, "Subtracting a negative number from a larger negative number yields a positive number"},
+		{0, []float64{6, 6, 15}, -27, "Subtracting multiple positive number from zero yields a negative number"},
+		{0, []float64{-22}, 22, "Subtracting a negative number from zero yields a positive number"},
+	}
+	for _, c := range testCases {
+		got := Subtract(c.a, c.b...)
+		if c.want != got {
+			t.Errorf("%s: want %f, got %f", c.name, c.want, got)
+		}
+	}
+}
+
+func TestMultiply(t *testing.T) {
```

Comment:
And because these tests are all so similar, we could test Add, Subtract, and Multiply all in one test, couldn't we?

Reply:
Brilliant. Done.